packages/seacas/libraries/ioss/src/Ioss_Utils.C

```diff
@@ -73,41 +73,71 @@ namespace {
   size_t MAX(size_t a, size_t b) { return b ^ ((a ^ b) & -static_cast<int>(a > b)); }
   size_t max_field_size = 0;
 
-  // Data space shared by most field input/output routines...
-  std::vector<char>    data;
-  std::vector<int>     data_int;
-  std::vector<int64_t> data_int64;
-  std::vector<double>  data_double;
-  std::vector<Complex> data_complex;
+  struct DataPool
+  {
+    // Data space shared by most field input/output routines...
+    std::vector<char>    data;
+    std::vector<int>     data_int;
+    std::vector<int64_t> data_int64;
+    std::vector<double>  data_double;
+    std::vector<Complex> data_complex;
 #ifdef SEACAS_HAVE_KOKKOS
-  Kokkos::View<char *>    data_view_char;
-  Kokkos::View<int *>     data_view_int;
-  Kokkos::View<int64_t *> data_view_int64;
-  Kokkos::View<double *>  data_view_double;
-  // Kokkos::View<Kokkos_Complex *> data_view_complex cannot be a global variable,
-  // Since Kokkos::initialize() has not yet been called. Also, a Kokkos:View cannot
-  // have type std::complex entities.
-  Kokkos::View<char **>    data_view_2D_char;
-  Kokkos::View<int **>     data_view_2D_int;
-  Kokkos::View<int64_t **> data_view_2D_int64;
-  Kokkos::View<double **>  data_view_2D_double;
-  // Kokkos::View<Kokkos_Complex **> data_view_2D_complex cannot be a global variable,
-  // Since Kokkos::initialize() has not yet been called. Also, a Kokkos:View cannot
-  // have type std::complex entities.
-  Kokkos::View<char **, Kokkos::LayoutRight, Kokkos::HostSpace>    data_view_2D_char_layout_space;
-  Kokkos::View<int **, Kokkos::LayoutRight, Kokkos::HostSpace>     data_view_2D_int_layout_space;
-  Kokkos::View<int64_t **, Kokkos::LayoutRight, Kokkos::HostSpace> data_view_2D_int64_layout_space;
-  Kokkos::View<double **, Kokkos::LayoutRight, Kokkos::HostSpace>  data_view_2D_double_layout_space;
-// Kokkos::View<Kokkos_Complex **, Kokkos::LayoutRight, Kokkos::HostSpace>
-// data_view_2D_complex_layout_space cannot be a global variable,
-// Since Kokkos::initialize() has not yet been called. Also, a Kokkos:View cannot
-// have type std::complex entities.
+    Kokkos::View<char *>    data_view_char;
+    Kokkos::View<int *>     data_view_int;
```

Comment:
Could it be possible for you only to have a single allocation, `data_view_char`, and alias all the other Views to its data? or do you really need separate allocations for separate types?  It just seems like you're keeping around a bunch of extra allocations.

Reply:
That is how the non-Kokkos version works.  I will take a look at modifying the Kokkos version to do this.

Reply:
It's even easier than with `std::vector`: just use the `Kokkos::View<char*>` as a memory pool, and give its pointer to each of the other `Kokkos::View`.  You can construct a nonowning `Kokkos::View` with a pointer and dimensions, like this:
```c++
Kokkos::View<char*> memoryPool ("memoryPool", poolSize);
Kokkos::View<int**> nonowning (memoryPool.data (), dim0, dim1);
```

Reply:
Thanks.  I will look into this and see what it takes (a little more involved than shown above, but should be doable in a similar manner...).  Can you approve the current changes and then I will work on the single allocation changes at a later time.